test/types/cptr/ddata_casts/c_ptr_ddata_cast_wrong_type.chpl

```diff
@@ -0,0 +1,10 @@
+use CTypes;
+
+var a : int = 4;
+
+// casting from non-matching element type: ok
+// Unsure if this is intentional. This test was added while ensuring the
+// c_void_ptr->c_ptr(void) replacement didn't affect anything, and this was
+// also possible before. Anna, July 2023
+var mycptr : c_ptr(int) = c_ptrTo(a);
+var myddata : _ddata(real) = mycptr : _ddata(real);
```

Comment:
I don't think that would do it; the `c_ptr`->`_ddata` cast definition requires that the `eltType`s are `==`. I suspect this is possible because of the `c_ptr(void)`->`_ddata` cast I just added, and `c_ptr(t)` of any element type can be coerced to `c_ptr(void)`. If that's correct, the reason this was allowed before the `c_void_ptr` replacement would have been similar, as we had a `c_ptr_void`->`_ddata` cast and any `c_ptr(t)` was coercible to `c_void_ptr`.